### ISSUE TITLE
Enhancements

### ISSUE THREAD

**Issue by derhier** [RainbowMiner/RainbowMiner#2603]
I got some enhancements:

Include a edit button in most of the Dashboard pages - OC profile page displays profiles greatly, but you need to go to json editor and do everything there - therefore OC Profile page is redundant, as if you want to make a change you would just go to the editor in the first place. 

Including Algo selection in OC profiles

Server Dashboard is lacking basic/any client controls:
-Server dashboard doesn't have any way of controlling paused clients
-dashboard doesn't have a way to reboot clients

Benchmarks are stored locally and can't be controlled by Server - if changes to OC profile and you want to update benchmarks,  you would need to go to each client and make the settings there. Very bad for scalability.

ssl option not in normal config-setup menu

DefaultOCprofile not in config-setup-devices page
default OC profile for card should be in the OCprofile-config.txt as it would be central, and you wouldn't need to go to clients and set it there individually -> scalability&ease of use

**Comment by whitindllc** [comment]
Having a button to push server config files to clients - noticed that OCprofiles didnt get pulled from clients sometimes after edit

i dont think that gets pushed down... wouldnt make sense to

**Comment by RainbowMiner** [comment]
Very good points. I will try to add these step by step. To supply clients with individual configurations, just create a folder with the exact client's miner name and put the individual config files there.

**Comment by derhier** [comment]
> Server Dashboard is lacking basic/any client controls: -Server dashboard doesn't have any way of controlling paused clients -dashboard doesn't have a way to reboot clients
> 
> why are the clients paused to begin with? they dont just pause themselves reboot the app? or the whole client machine?
> 
> the OC thing is better controlled with msi afterburner for some of us... others use the OC profiles... i never had any luck with it

OC Profiles work like a charm. Afterburner works on Windows, and windows is not sustainable beyond a few rigs. 

I set up a pause in miner and left the ssh. Wanted to control it in the server and just noticed the missing controls. No biggie. - just mentioned due to scalability. If you have 500 rigs, you can't babysit them individually.



About the pushing - yeah, it usually does. Just noticed that it didn't for me in a situation, so something to schedule it manually might be a good thing?

**Comment by derhier** [comment]
> Very good points. I will try to add these step by step. To supply clients with individual configurations, just create a folder inside the `./Config` folder with the exact client's miner name and put the individual config files there.

Yeah, I see a lot of potential in RainbowMiner. Looking to apply it to our operation. 

I didn't want to supply individual config. I set the central OC and one rig wasn't pulling it, until the 3rd restart or so. A button which schedules a config update to all clients would be very helpful for such occurrences, I guess. Not sure if i was inpatient or user error, tho.
